contracts/0.8.25/vaults/StakingVault.sol

```diff
@@ -0,0 +1,369 @@
+import {ERC1967Utils} from "@openzeppelin/contracts-v5.0.2/proxy/ERC1967/ERC1967Utils.sol";
+import {VaultHub} from "./VaultHub.sol";
+import {IReportReceiver} from "./interfaces/IReportReceiver.sol";
+import {IStakingVault} from "./interfaces/IStakingVault.sol";
+import {IBeaconProxy} from "./interfaces/IBeaconProxy.sol";
+import {VaultBeaconChainDepositor} from "./VaultBeaconChainDepositor.sol";
+
+/**
+ * @title StakingVault
+ * @author Lido
+ * @notice A staking contract that manages staking operations and ETH deposits to the Beacon Chain
+ * @dev
+ *
+ * ARCHITECTURE & STATE MANAGEMENT
+ * ------------------------------
+ * The vault uses ERC7201 namespaced storage pattern with a main VaultStorage struct containing:
+ * - report: Latest metrics snapshot (valuation and inOutDelta at time of report)
+ * - locked: Amount of ETH that cannot be withdrawn (managed by VaultHub)
+ * - inOutDelta: Running tally of deposits minus withdrawals since last report
+ *
+ * CORE MECHANICS
+ * -------------
+ * 1. Deposits & Withdrawals
+ *    - Owner can deposit ETH via fund()
+ *    - Owner can withdraw unlocked ETH via withdraw()
+ *    - All deposits/withdrawals update inOutDelta
+ *    - Withdrawals are only allowed if vault remains healthy
+ *
+ * 2. Valuation & Health
+ *    - Total value = report.valuation + (current inOutDelta - report.inOutDelta)
+ *    - Vault is "healthy" if total value >= locked amount
+ *    - Unlocked ETH = max(0, total value - locked amount)
+ *
+ * 3. Beacon Chain Integration
+ *    - Can deposit validators (32 ETH each) to Beacon Chain
+ *    - Withdrawal credentials are derived from vault address
+ *    - Can request validator exits when needed by emitting the event,
+ *      which acts as a signal to the operator to exit the validator,
+ *      Triggerable Exits are not supported for now
+ *
+ * 4. Reporting & Updates
+ *    - VaultHub periodically updates report data
+ *    - Reports capture valuation and inOutDelta at the time of report
+ *    - VaultHub can increase locked amount outside of reports
+ *
+ * 5. Rebalancing
+ *    - Owner or VaultHub can trigger rebalancing when unhealthy
+ *    - Moves ETH between vault and VaultHub to maintain health
+ *
+ * ACCESS CONTROL
+ * -------------
+ * - Owner: Can fund, withdraw, deposit to beacon chain, request exits
+ * - VaultHub: Can update reports, lock amounts, force rebalance when unhealthy
+ * - Beacon: Controls implementation upgrades
+ *
+ * SECURITY CONSIDERATIONS
+ * ----------------------
+ * - Locked amounts can only increase outside of reports
+ * - Withdrawals blocked if they would make vault unhealthy
+ * - Only VaultHub can update core state via reports
+ * - Uses ERC7201 storage pattern to prevent upgrade collisions
+ * - Withdrawal credentials are immutably tied to vault address
+ *
+ */
+contract StakingVault is IStakingVault, IBeaconProxy, VaultBeaconChainDepositor, OwnableUpgradeable {
```

Comment:
we are violating LIP-XX about initialization approach :warning: let's make it explicit